gosling/api.py

```diff
@@ -265,24 +261,7 @@ def view(self, **kwargs) -> View:
 
 def overlay(*tracks: Union[Track, View], **kwargs) -> View:
     """Compose an overlaid view from multiple tracks or overliad tracks"""
-    # Overlay requires a `width` and `height`. Check if provided as kwargs, otherwise
-    # eagerly grab the width/height from the tracks, otherwise use defaults.
-    width = kwargs.pop(
-        "width", next((t.width for t in tracks if t.width != Undefined), DEFAULT_WIDTH)
-    )
-
-    height = kwargs.pop(
-        "height",
-        next((t.height for t in tracks if t.height != Undefined), DEFAULT_HEIGHT),
-    )
-
-    # TODO: Gosling.js doesn't respect the parent width/height if defined in children.
-    # This is a hack to ensure the children respect the view-level config.
-    tracks = tuple(t.properties(width=Undefined, height=Undefined) for t in tracks)
```

Comment:
Let me know if you think it makes sense to support this in Gosling.

Currently, my intention was to remove `width` and `height` from the individual overlaid tracks, i.e. disallowing the `width` and `height` since they are not used at all: 

```ts
{ alignment: 'overlay', tracks: [{ 
  ..., width: 100 // error
}]}}
```

This is implemented in gosling.schema.ts:

https://github.com/gosling-lang/gosling.js/blob/master/src/core/gosling.schema.ts#L399-L407
```ts
export type OverlaidTrack = Partial<SingleTrack> & {
    overlay: Partial<Omit<SingleTrack, 'height' | 'width' | 'layout' | 'title' | 'subtitle'>>[];
};
```

However, since `Omit` is not supported in `ts-json-schema-generator`, this is not included on our `gosling.schema.json`.

But, in the context of `gos` this more scoped schema could make it confusing to users since they cannot overlay tracks if width or height is defined.

Reply:
Oh, I think I misinterpreted. I think Gosling currently does not respect children's width/height instead of 'the parent width/height`.